components/accelerator/nvidia/infiniband/threshold/threshold.go

```diff
@@ -0,0 +1,122 @@
+// Package threshold contains the threshold for the infiniband component.
+package threshold
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/leptonai/gpud/components"
+	"github.com/leptonai/gpud/components/common"
+	events_store "github.com/leptonai/gpud/components/db"
+)
+
+// Defines the minimum number of ports and the expected rate in Gb/sec.
+type Threshold struct {
```

Comment:
Do we support syncing node hardware spec like ib threshold from server?  /cc @cardyok

Reply:
Yes

https://github.com/leptonai/gpud/blob/9fe57e679884c752af3328d1b60ee0c2729eae03/internal/session/serve.go#L170-L183

(Current issue is this update will reset on GPUd restart)